### PR TITLE
Upgrade Micrometer from 1.12.4 to 1.13.0

### PR DESCRIPTION
**Pull request by yeikel**
https://github.com/micrometer-metrics/micrometer/releases/tag/v1.13.0

Migration guide : https://github.com/micrometer-metrics/micrometer/wiki/1.13-Migration-Guide

### CODE REVIEW

**Comment by tsegismont** [comment]
Thanks for starting this effort @yeikel ! I'll start reviewing when the compilation issues are resolved.

File: src/main/java/io/vertx/micrometer/impl/PrometheusScrapingHandlerImpl.java [hunk 1]
```diff
@@ -74,7 +72,7 @@ public void handle(RoutingContext rc) {
 
   private static void scrapeAndReply(RoutingContext rc, PrometheusMeterRegistry prometheusMeterRegistry) {
     rc.response()
-      .putHeader(CONTENT_TYPE, CONTENT_TYPE_004)
+      .putHeader(CONTENT_TYPE, "text/plain; version=0.0.4; charset=utf-8")
```
Comment (yeikel) [inline src/main/java/io/vertx/micrometer/impl/PrometheusScrapingHandlerImpl.java:75]: As far as I can tell, to get access to this constant, we'd need to add the `prometheus-metrics-exposition-formats` dependency to the classpath 

https://github.com/prometheus/client_java/blob/c0a38270c021e4d63998a98a68bacf2b6f32ace5/prometheus-metrics-exposition-formats/src/main/java/io/prometheus/metrics/expositionformats/PrometheusTextFormatWriter.java#L35

In `micrometer-metrics` this is now a private enum https://github.com/micrometer-metrics/micrometer/blob/37883fa6fb4a6d3f83d01f6b53101cc9f52b3f78/implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/PrometheusMeterRegistry.java#L617 

Should we have our constant or is inlining it like this ok?

**Comment by yeikel** [comment]
@tsegismont I resolved the build issues and added a question above. Please let me know what you think :)

**Comment by vietj** [comment]
I've seen that this release of micrometer has breaking changes (package name)

File: src/main/java/io/vertx/micrometer/impl/PrometheusScrapingHandlerImpl.java (hunk 1 above)
Comment (tsegismont) [reply src/main/java/io/vertx/micrometer/impl/PrometheusScrapingHandlerImpl.java:75]: Yeah, I had noticed this when trying out the candidate release of Micrometer. It is fine like this.

**Comment by tsegismont** [comment]
> I've seen that this release of micrometer has breaking changes (package name)

Yes it does, we need a mention to this in the breaking changes page (link to the Micrometer release notes)

**Review by tsegismont — APPROVED** [review]
@yeikel thank you!

**Pull request merged**